rotkehlchen/history/events.py

```diff
@@ -300,39 +302,50 @@ def get_history(
         history: List['AccountingEventMixin'] = []
         empty_or_error = ''
 
-        def populate_history_cb(
-                trades_history: List[Trade],
-                margin_history: List[MarginPosition],
-                result_asset_movements: List[AssetMovement],
-                exchange_specific_data: Any,
-        ) -> None:
-            """This callback will run for succesfull exchange history query
-
-            We don't include ledger actions here since we simply gather all of them at the end
-            """
-            history.extend(trades_history)
-            history.extend(margin_history)
-            history.extend(result_asset_movements)
-
-            if exchange_specific_data:
-                pass  # this used to be only for polo loans -- removed now. TODO: Think if needed
-
         def fail_history_cb(error_msg: str) -> None:
             """This callback will run for failure in exchange history query"""
             nonlocal empty_or_error
             empty_or_error += '\n' + error_msg
 
         for exchange in self.exchange_manager.iterate_exchanges():
             self.processing_state_name = f'Querying {exchange.name} exchange history'
-            exchange.query_history_with_callbacks(
+            exchange.query_history_with_fail_callback(
                 # We need to have history of exchanges since before the range
                 start_ts=Timestamp(0),
                 end_ts=end_ts,
-                success_callback=populate_history_cb,
                 fail_callback=fail_history_cb,
             )
             step = self._increase_progress(step, total_steps)
 
+        # Query all trades, asset movements and margin positions from the DB.
+        # Query all of them for all locations (both exchanges and other locations).
+        with self.db.conn.read_ctx() as cursor:
+            # Include all trades
```

Comment:
few comments here which count for all steps under the cursor:

1. Since you have a string name of what you do there is no need for a comment in each step.
2. The processing state name in each step is not correct. The time you query from the DB is not the step time. Reading from the DB is near instant. The steps are to appear in the PnL report view. So this step should appear only for when querying the actual trade history from remotes.
3. Same for the increase step

Reply:
Why did you remove the steps? I told you where they should go in my last comment.

> the steps are to appear in the PnL report view. So this step should appear only for when querying the actual trade history from remotes.

So you should probably add as many steps as are required in the callback.

Reply:
I am not sure that I get you. Which callback do you mean?

> So this step should appear only for when querying the actual trade history from remotes.

There already are steps when querying exchanges. What other location do you mean?